packages/contracts/contracts/libraries/NodeReader.sol

```diff
@@ -173,6 +173,7 @@ library NodeReader {
             // TODO(chokobole): Do the length check as much as possible at once and read the bytes.
```

Comment:
With this fix, can we remove this `TODO`?

Reply:
If we add logic at the end to verify that the total number of bytes processed is equal to _proof.length, we can delete the Since the read* functions check for length, I think we can delete them. 
@chokobole Can I delete it?

Reply:
Yes! only if you add the logic to verify the length you read is equal to the given length .